src/ExtensionConfigHandlerInterface.php

```diff
@@ -107,6 +108,47 @@ public function getStorageComparer(array $source_dirs, $subdir = InstallStorage:
   public function importFromComparer(StorageComparerInterface $storage_comparer);
 
   /**
+   * Get the config dependancies for an extension.
+   *
+   * For the config currently exported to a project, find the config
+   * dependencies required for it to work.
+   *
+   * @param string $extension
+   *   The machine name of the project to get the config dependencies for.
+   * @return array
+   *   An array of config the extension depends on.
+   */
+  public function getConfigDependancies($extension);
+
+  /**
+   * Suggest config dependants to export.
+   *
+   * For the config listed in a projects .info.yml, find other config that is
+   * dependant upon it, but which is not:
+   *  - Itself a dependency of the config listed in cm_config_tools.export
+   *  - Already included explicitly in cm_config_tools.export
+   *  - Explicitly ignored in the cm_config_tools.ignore
+   *
+   * @param string $extension
+   *   The machine name of the project to find dependant config for.
+   * @return array
+   *   An array of config that depends on config currently exported.
+   */
+  public function getDependentConfigSuggestions($extension);
```

Comment:
Note that in another comment I also suggest that we rename this to be clear that its dealing with a whole extension, so in that case, I suggest calling this getExtensionDependants() (or getExtensionSuggestedConfig / getSuggestedExtensionConfig )?